### DISCUSSION TITLE
Self Hosting, External Db and realtime?

### DISCUSSION THREAD

**Discussion by shankar-narayanan-d** [supabase/supabase#11912] [Questions]
My use case is, I have a local postgresql db. I need realtime updates for my clients when there is an updation in db.

So I followed the [official docs](https://supabase.com/docs/guides/self-hosting/docker)
Steps followed:

1. git clone --depth 1 https://github.com/supabase/supabase
2. cd supabase/docker
3. cp .env.example .env
4. Then I generated jwt, anon and service key using [key generator](https://supabase.com/docs/guides/self-hosting#api-keys)
5. I replcaed these keys in .env file and volumes/api/kong.yml file
6. replaced the following in .env file with my local db credentials:
    POSTGRES_PASSWORD=MY DB PASSWORD
    POSTGRES_HOST=host.docker.internal
    POSTGRES_DB=DB_NAME
    POSTGRES_USER=USER_NAME
    POSTGRES_PORT=5432
7. Then commented out the db section of the docker-compose file, and remove any instances where the services depends_on the db image.
8. Run all db/init-scripts with postgres superuser role. [files](https://github.com/supabase/postgres/tree/develop/migrations/db/init-scripts)
9. Run all db/migrations with supabase_admin superuser role. [files](https://github.com/supabase/postgres/tree/develop/migrations/db/migrations)
10. Finally run docker compose up in supabase/docker directory.

Error:
{"code":"PGRST000","details":"FATAL: could not access file \"safeupdate\": No such file or directory\n","hint":null,"message":"Database connection error. Retrying the connection."}

and

12:53:58.660 project=realtime-dev external_id=realtime-dev [error] Prepare replication error: %Postgrex.Error{message: nil, postgres: %{code: :undefined_file, file: "dfmgr.c", line: "210", message: "could not access file \"wal2json\": No such file or directory", pg_code: "58P01", routine: "internal_load_library", severity: "ERROR", unknown: "ERROR"}, connection_id: 21885, query: nil}

Can some help me on this?

**Comment by cristian-milea** [comment]
I have similar questions and yours not being answered for almost 2 months got me thinking... "is this self-hosted Supabase dream abandoned?"

**Comment by kiyasov** [comment]
Solving the problem, install https://github.com/eradman/pg-safeupdate 
and sql 
`load 'safeupdate';`
